Change to 'dnf' as package manager for Fedora 22->

mbologna opened the pull request:
If you are installing Saltstack on Fedora 22 onwards, the default package manager
is dnf (instead of yum).
This patch decide which package manager to use based on Fedora version and the
availability of dnf on the current system. If found, uses dnf to install packages
needed by saltstack.

Using this patch final user will not see yum deprecation warnings like:
Yum command has been deprecated, redirecting to '/usr/bin/dnf install -y yum-utils..
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

s0undt3ch commented:
What happened to the symlinks?

mbologna commented:
@s0undt3ch you're right. Corrected and re-push'ed -f

mbologna committed:
Change to 'dnf' as package manager for Fedora 22->

If you are installing Saltstack on Fedora 22 onwards, the default package manager
is dnf (instead of yum).
This patch decide which package manager to use based on Fedora version and the
availability of dnf on the current system. If found, uses dnf to install packages
needed by saltstack.

Using this patch final user will not see yum deprecation warnings like:
Yum command has been deprecated, redirecting to '/usr/bin/dnf install -y yum-utils..
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

jfindlay commented on bootstrap-salt.sh:
```diff
@@ -2626,7 +2626,17 @@ install_debian_check_services() {
+
+FEDORA_PACKAGE_MANAGER="yum"
+
+fedora_get_package_manager() {
```
@mbologna, would you mind renaming this to `__fedora_get_package_manager`?  Thanks.

mbologna replied:
@jfindlay done in 10bcdf3

mbologna committed:
Renamed to __fedora_get_package_manager

Renamed newly introduced fedora_get_package_manager() to
__fedora_get_package_manager()

jfindlay commented on bootstrap-salt.sh:
```diff
@@ -2626,7 +2626,17 @@ install_debian_check_services() {
+FEDORA_PACKAGE_MANAGER="yum"
+
+__fedora_get_package_manager() {
+  if [ "$DISTRO_MAJOR_VERSION" -lt 22 ] || [ "$(which dnf)" != "" ]; then
```
@mbologna, sorry for another comment, but shouldn't this line be
`if [ "$DISTRO_MAJOR_VERSION" -ge 22 ] || [ "$(which dnf)" != "" ]; then`

mbologna replied:
Yep, you're right. Correcting it right now and pushing.

mbologna committed:
Correcting: from 22 onwards

jfindlay commented:
Thanks, @mbologna.
